Client/Application/AppDelegate+PushNotifications.swift

```diff
@@ -118,10 +118,27 @@ extension AppDelegate {
             notificationAllowed = UserDefaults.standard.bool(forKey: PrefsKeys.Notifications.SyncNotifications)
         }
 
-        if notificationAllowed {
-            RustFirefoxAccounts.shared.pushNotifications.didRegister(withDeviceToken: deviceToken)
-        } else {
+        guard notificationAllowed else {
             RustFirefoxAccounts.shared.pushNotifications.disableNotifications()
+            return
+        }
+
+        guard LegacyFeatureFlagsManager.shared.isFeatureEnabled(.autopushFeature, checking: .buildOnly) else {
+            RustFirefoxAccounts.shared.pushNotifications.didRegister(withDeviceToken: deviceToken)
+            return
+        }
+        // We set this here because the NotificationService can't depend on the nimbus setting yet
+        // and will read the profile pref directly
+        LegacyFeatureFlagsManager.shared.set(feature: .autopushFeature, to: true)
+        Task {
+            do {
+                let autopush = try await Autopush(files: profile.files)
+                try await autopush.updateToken(withDeviceToken: deviceToken)
+                let fxaSubscription = try await autopush.subscribe(scope: RustFirefoxAccounts.pushScope)
+                RustFirefoxAccounts.shared.pushNotifications.updatePushRegistration(subscriptionResponse: fxaSubscription)
+            } catch let e {
+                logger.log("Failed to update push registartion", level: .warning, category: .setup, description: e.localizedDescription)
```

Comment:
Preferrably we use variables names that are longer than 1-2 letters
```suggestion
            } catch let error {
                logger.log("Failed to update push registartion",
                           level: .warning,
                           category: .setup,
                           description: error.localizedDescription)
```